qemu/tests/win_irq_check.py

```diff
@@ -38,13 +55,40 @@ def irq_check(session, device_name, devcon_folder, timeout):
             get_irq_cmd = '%sdevcon.exe resources @"%s" | find "IRQ"' % (devcon_folder,
                                                                          hwid)
             output = session.cmd_output(get_irq_cmd)
-            irq_value = re.split(r':+', output)[1]
-            if ctypes.c_int32(int(irq_value)).value < 0:
-                test.fail("%s's irq is not correct." % device_name, logging.info)
+            irq_list = output.strip().splitlines()
+            irq_nums = len(irq_list)
+            for irq in irq_list:
+                irq_value = re.split(r':+', irq)[1]
+                if irq_nums != 1:
+                    vectors = get_vectors_fqtree()
+                    if irq_nums != int(vectors):
+                        test.fail("%s's irq nums not equal to vectors." % device_name)
+                    if ctypes.c_int32(int(irq_value)).value >= 0:
+                        test.fail("%s's irq is not correct." % device_name)
+                else:
+                    if ctypes.c_int32(int(irq_value)).value < 0:
+                        test.fail("%s's irq is not correct." % device_name)
+
+    def disable_msi_fguest(session):
+        """
+        Disable MSI from guest.
+        """
+        hwids = win_dev.get_hwids(session, device_name, devcon_folder, timeout)
+        disable_msi_cmd = params["disable_msi_cmd"] % hwids[0]
+        session.cmd(disable_msi_cmd)
+
+    def enable_msi_fguest(session):
+        """
+        Enable MSI from guest.
+        """
+        hwids = win_dev.get_hwids(session, device_name, devcon_folder, timeout)
+        enable_msi_cmd = params["enable_msi_cmd"] % hwids[0]
+        session.cmd(enable_msi_cmd)
```

Comment:
The above two functions can be one, just transfer xxx_msi_cmd.

Reply:
Thanks, updated the code, please help to review it~
Thanks a lot~